x-pack/legacy/plugins/file_upload/public/util/file_parser.js

```diff
@@ -7,39 +7,106 @@
 import _ from 'lodash';
 import { geoJsonCleanAndValidate } from './geo_json_clean_and_validate';
 import { i18n } from '@kbn/i18n';
+const oboe = require('oboe');
 
-export async function readFile(file) {
-  const readPromise = new Promise((resolve, reject) => {
-    if (!file) {
-      reject(new Error(i18n.translate(
-        'xpack.fileUpload.fileParser.noFileProvided', {
-          defaultMessage: 'Error, no file provided',
-        })));
+// In local testing, performance improvements leveled off around this size
+export const FILE_BUFFER = 1024000;
+let featuresProcessed = 0;
```

Comment:
featuresProcessed is not getting reset when reading additional files.

Reply:
Seems a little weird to have a global like this when there could be multiple concurrent executions of parseFile since its async. How about moving featuresProcessed into the scope of parseFile?

Reply:
Then it wouldn't be visible where it's incremented on line 25. I'll take another look though and see if there's a smarter way to handle it

Reply:
I think there's room for some follow-up work in a separate issue/pr to give feedback on whether features passed `cleanAndValidate`, which ones were modified due to bad geometries, etc. I've added in some code to localize the variable and set us up for more feature processing stats feedback